diff -Naur a/Makefile b/Makefile
--- a/Makefile	2021-04-16 16:30:35.302171000 +0200
+++ b/Makefile	2021-04-16 16:35:16.947736909 +0200
@@ -13,13 +13,13 @@
 SHELL := /bin/bash
 CC=emcc
 CXX=em++
-OPTFLAGS=-O2
+OPTFLAGS=-O3
 CFLAGS=$(OPTFLAGS) -g -I$(PYTHONINCLUDE) -Wno-warn-absolute-paths
 CXXFLAGS=$(CFLAGS) -std=c++14
 
 
 LDFLAGS=\
-	-O2 \
+	-O3 \
 	-s MODULARIZE=1 \
 	$(CPYTHONROOT)/installs/python-$(PYVERSION)/lib/libpython$(PYMINOR).a \
 	-s "BINARYEN_METHOD='native-wasm'" \
diff -Naur a/pyodide_build/buildall.py b/pyodide_build/buildall.py
--- a/pyodide_build/buildall.py	2020-12-25 22:05:54.000000000 +0100
+++ b/pyodide_build/buildall.py	2021-04-16 16:39:34.915644083 +0200
@@ -72,10 +72,15 @@
                     stderr=subprocess.STDOUT,
                 )
 
-        with open(self.pkgdir / "build.log", "r") as f:
-            shutil.copyfileobj(f, sys.stdout)
+        try:
+            p.check_returncode()
+        except subprocess.CalledProcessError:
+            print(f"Error building {self.name}. Printing build logs.")
 
-        p.check_returncode()
+            with open(self.pkgdir / "build.log", "r") as f:
+                shutil.copyfileobj(f, sys.stdout)
+
+            raise
 
         if not self.library:
             shutil.copyfile(
diff -Naur a/pyodide_build/common.py b/pyodide_build/common.py
--- a/pyodide_build/common.py	2021-04-16 16:30:30.090275000 +0200
+++ b/pyodide_build/common.py	2021-04-16 16:35:29.200023229 +0200
@@ -7,7 +7,7 @@
 # fmt: off
 DEFAULTLDFLAGS = " ".join(
     [
-        "-O2",
+        "-O3",
         "-s", "BINARYEN_METHOD='native-wasm'",
         "-Werror",
         "-s", "EMULATED_FUNCTION_POINTERS=1",
